x-pack/plugin/security/src/test/java/org/elasticsearch/test/NativeRealmIntegTestCase.java

```diff
@@ -68,6 +70,16 @@ protected boolean addMockHttpTransport() {
         return templates;
     }
 
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        Settings.Builder builder = Settings.builder().put(super.nodeSettings(nodeOrdinal));
+        // we are randomly running a large number of nodes in these tests so we limit the number of worker threads
+        // since the default of 2 * CPU count might use up too much direct memory for thread-local direct buffers for each node's
+        // transport threads
+        builder.put(Netty4Transport.WORKER_COUNT.getKey(), random().nextInt(3) + 1);
```

Comment:
We use the same randomization for worker count in other tests so I picked it here as well ... no special reason for the parameter choice other than that.